File: Practice/Task5 perl/02/task2.pl
#!/usr/bin/perl
if ($ARGV[0] eq "--help")
{
	print "Usage: task2.pl DIR DIR_TO *.EXT\n";
	print "Copy all files that have EXT extention from DIR to DIR_TO\n";	
}
elsif ($ARGV[0] eq "--test")
{
	system("./test2.bash");
}
elsif (scalar @ARGV != 3)
{
	print "task2.pl: missing file operands\n";
	print "Try 'task2.pl --help' for more information\n";
}
else
{
	use File::Find::Rule;
	use File::Copy;
	my @files = File::Find::Rule->file()
                     		    ->name($ARGV[2])
                           	    ->in($ARGV[0]);
	for my $file (@files)
	{
		copy($file, $ARGV[1]) or die "Unable to copy: $!";	
	}
}
	

File: Practice/Task2 gcc/2/task2.cpp
#include <cstdlib>
#include <iostream>
#include <fstream>
#include <filesystem>
namespace fs = std::filesystem;
int main(int argc, char **argv)
{
	if(std::string(argv[1]) == std::string("--help"))
	{
		std::cout << "Usage ./task2 DIR DIR_TO .EXT" << std::endl;
		std::cout << "Copy all files that have EXT extention from DIR to DIR_TO" << std::endl; 
	}	
	else if (std::string(argv[1]) == std::string("--test"))
	{
		system("./test2.bash");
	}
	else if (argc != 4)
	{	
		std::cout << "task2: missing file operands" << std::endl;
		std::cout << "Try './task2 --help' for more information" << std::endl;
	}
	else 
	{
		for (const auto & p : fs::recursive_directory_iterator(argv[1])) 
    		{	
     			fs::path fileToCopy = p; 
      			if (fileToCopy.extension() == argv[3])
  			{
      				fs::path target = argv[2] / fileToCopy.filename(); 
        			fs::copy_file(p,target,fs::copy_options::overwrite_existing); 
				//static_cast<void>(std::system("tree"));
			}
		}	
	}
	return 0;
}
